tests/limma/test_lmFit.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+import numpy as np
+import pandas as pd
+from patsy import dmatrix
+from scipy.stats import norm
+
+from inmoose.utils import Factor
+from inmoose.limma import lmFit, nonEstimable
+
+
+class Test(unittest.TestCase):
+    def setUp(self):
+        y = norm.rvs(size=(10, 6), scale=0.3, random_state=42)
+        y[0, :2] += 2
+        self.y = y
+        group = Factor([1, 1, 1, 2, 2, 2])
+        self.design = dmatrix("0+group")
+        self.contrast_matrix = pd.DataFrame(
+            {"First3": [1, 0], "Last3": [0, 1], "Last3-First3": [-1, 1]}
+        )
+
+    def test_nonEstimable(self):
+        d = dmatrix("x", data={"x": ["A", "B"]})
+        self.assertEqual(nonEstimable(d), None)
+        d = dmatrix("x+y", data={"x": ["A", "B"], "y": ["B", "A"]})
+        self.assertEqual(nonEstimable(d), ["x[T.B]"])
+
+    def test_lmFit(self):
+        with self.assertRaisesRegex(
+            ValueError, expected_regex="the correlation must be set"
+        ):
+            lmFit(self.y, self.design, ndups=2)
+
+        # fit = lmFit(self.y, self.design, ndups=2, correlation=0.9)
+        # coef_ref = np.array(
+        #    [
+        #        [1.144165577, 0.03340760],
+        #        [-0.212545740, -0.14219338],
+        #        [-0.017368967, -0.09863220],
+        #        [-0.248713327, 0.04916677],
+        #        [0.001386212, 0.02736341],
+        #    ]
+        # )
+        # self.assertTrue(np.allclose(fit.coefficients, coef_ref))
```

Comment:
not sure why this test is in a comment, there's a problem with the `ndups` + `correlation` parameters?

Reply:
Maybe it’s because the option is in another PR, isn't it? So it can't work before merging the PR "duplicates from limma"

Reply:
I started writing the test before we decided not to implement the duplicated case.
I thought I'd leave the values here to easily extend the test if and when we implement the duplicated case.